testsuite/pytests/sli2py_regressions/test_issue_2629.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# test_issue_2629.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Regression test for Issue #2629 (GitHub).
+
+The issue was that ``DumpLayerConnections`` failed when a source layer was
+connected to more than one target layer. The test ensures that this is no
+longer the case.
+"""
+
+import pytest
+
+import nest
+
+
+@pytest.fixture(autouse=True)
+def reset():
+    nest.ResetKernel()
+
+
+def test_dump_layer_connections_multiple_targets(tmp_path):
```

Comment:
I'd suggest to split this into a fixture building the network and then two separate tests.

I assume the tmp_path fixture automatically deletes the files after the test?

Reply:
The test has been updated as you suggested in https://github.com/nest/nest-simulator/pull/2920/commits/398ec15a3d76e54c0467905512a40b7b66b5c275.

No, `tmp_path` directories are not deleted immediately after the test. According to the [Pytest documentation](https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-default-base-temporary-directory), the last 3 temporary directories are kept. I guess this is for having artifacts available in case tests fail.